include/simics/dmllib.h

```diff
@@ -2113,8 +2113,10 @@ _set_device_member(attr_value_t val,
                             &dimension_strides[1],
                             remaining_dimensions-1,
                             setter);
-                        // currently, generated deserialized cannot fail
-                        ASSERT(status == Sim_Set_Ok);
+                        // If deserialization fails, bail out.
+                        if (unlikely(status != Sim_Set_Ok)) {
```

Comment:
Ideally, a failed setter should have no effect, i.e. we should split _set_device_member in two stages, a side-effect free `is_legal()` plus a never failing `set()`. However, this is not a strict requirement since we already break this rule in arrays of attributes or connects.